### PR TITLE
chore: poll the package policies to detect if package is in policy

### PR DESCRIPTION
**Pull request by mdelapenya**
- chore: log error
- chore: lowercase error message in fmt.Errorf
- chore: log when the package was found in the policy
- chore: poll the policies for the package datastream to match
- chore: simpler if condition

## What does this PR do?
It creates a wait with backoff strategy for checking if the package is in the policy. Before these changes, it listed all package policies, looped through them, and verified that the policy ID matched AND the package name/title was equals to the argument.

## Why is it important?
With these changes we are polling the package policies to identify if the tests happened too fast and for that reason the flakiness: it could be the case the tests performed the verification at a moment of time that was too early for Fleet to provide the right values.

That would explain the flakiness in which some scenarios sometimes pass sometimes not.

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the Unit tests (`make unit-test`), and they are passing locally
- [ ] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)

## Related issues

- Relates to all the issues labelled with `flaky-test`. If these changes fixes the flakiness, I'd close all those issues

### CODE REVIEW

**Comment by mdelapenya** [comment]
@jlind23 we expect this PR helps in detect the flakiness for the system-integration scenarios. As described above, it could be the case that test tests were running too fast (automated) and Fleet was not updating the state with the same speed (reasonably expected?)

**Comment by mdelapenya** [comment]
Mmm weird:

> [2022-08-03T08:00:39.877Z] time="2022-08-03T08:00:39Z" level=trace msg="Kibana API Query" body=null headers="[]" method=GET url="http://18.225.37.65:5601/api/fleet/package_policies"
[2022-08-03T08:00:39.877Z] time="2022-08-03T08:00:39Z" level=warning msg="Package not found in policy" elapsedTime=36.563034ms package=system policyID=04b11590-1302-11ed-8ab7-3dd598999d32 retries=1

but the JSON response includes it as the first item (the e2e code looks up the integration by name, then by title and finally by package name) https://github.com/elastic/e2e-testing/blob/ac9a3263e25833122578119e864022457d369f26/internal/kibana/integrations.go#L204

The JSON response:

```json
{
    "items": [
        {
            "id": "e7869804-6608-452e-8f33-59b008108f57",
            "version": "WzMzMjQsMV0=",
            "name": "system-821ef8c8-6e64-4552-a301-51246ff91c46",
            "namespace": "default",
            "description": "System",
            "package": {
                "name": "system",
                "title": "System",
                "version": "1.16.2"
            },
            "enabled": true,
            "policy_id": "5c3e7f30-1304-11ed-8ab7-3dd598999d32",
            "output_id": "",
            "inputs": [
                {
                    "streams": [
                        {
                            "compiled_stream": {
                                "exclude_files": [
                                    ".gz$"
                                ],
                                "paths": [
                                    "/var/log/auth.log*",
                                    "/var/log/secure*"
                                ],
                                "multiline": {
                                    "pattern": "^\\s",
                                    "match": "after"
                                },
                                "processors": [
                                    {
                                        "add_locale": null
                                    }
                                ]
                            },
                            "data_stream": {
                                "type": "logs",
                                "dataset": "system.auth"
                            },
                            "id": "logfile-system.auth-e7869804-6608-452e-8f33-59b008108f57",
                            "vars": {
                                "paths": {
                                    "type": "text",
                                    "value": [
                                        "/var/log/auth.log*",
                                        "/var/log/secure*"
                                    ]
                                }
                            },
                            "enabled": true
                        },
                        {
                            "compiled_stream": {
                                "exclude_files": [
                                    ".gz$"
                                ],
                                "paths": [
                                    "/var/log/messages*",
                                    "/var/log/syslog*"
                                ],
                                "multiline": {
                                    "pattern": "^\\s",
                                    "match": "after"
                                },
                                "processors": [
                                    {
                                        "add_locale": null
                                    }
                                ]
                            },
                            "data_stream": {
                                "type": "logs",
                                "dataset": "system.syslog"
                            },
                            "id": "logfile-system.syslog-e7869804-6608-452e-8f33-59b008108f57",
                            "vars": {
                                "paths": {
                                    "type": "text",
                                    "value": [
                                        "/var/log/messages*",
                                        "/var/log/syslog*"
                                    ]
                                }
                            },
                            "enabled": true
                        }
                    ],
                    "type": "logfile",
                    "enabled": true
                },
                {
                    "streams": [
                        {
                            "data_stream": {
                                "type": "logs",
                                "dataset": "system.application"
                            },
                            "id": "winlog-system.application-e7869804-6608-452e-8f33-59b008108f57",
                            "enabled": false
                        },
                        {
                            "data_stream": {
                                "type": "logs",
                                "dataset": "system.security"
                            },
                            "id": "winlog-system.security-e7869804-6608-452e-8f33-59b008108f57",
                            "enabled": false
                        },
                        {
                            "data_stream": {
                                "type": "logs",
                                "dataset": "system.system"
                            },
                            "id": "winlog-system.system-e7869804-6608-452e-8f33-59b008108f57",
                            "enabled": false
                        }
                    ],
                    "type": "winlog",
                    "enabled": false
                },
                {
                    "streams": [
                        {
                            "compiled_stream": {
                                "metricsets": [
                                    "core"
                                ],
                                "core.metrics": [
                                    "percentages"
                                ]
                            },
                            "data_stream": {
                                "type": "metrics",
                                "dataset": "system.core"
                            },
                            "id": "system/metrics-system.core-e7869804-6608-452e-8f33-59b008108f57",
                            "vars": {
                                "period": {
                                    "type": "text",
                                    "value": "10s"
                                },
                                "core.metrics": {
                                    "type": "text",
                                    "value": [
                                        "percentages"
                                    ]
                                }
                            },
                            "enabled": true
                        },
                        {
                            "compiled_stream": {
                                "period": "10s",
                                "cpu.metrics": [
                                    "percentages",
                                    "normalized_percentages"
                                ],
                                "metricsets": [
                                    "cpu"
                                ]
                            },
                            "data_stream": {
                                "type": "metrics",
                                "dataset": "system.cpu"
                            },
                            "id": "system/metrics-system.cpu-e7869804-6608-452e-8f33-59b008108f57",
                            "vars": {
                                "period": {
                                    "type": "text",
                                    "value": "10s"
                                },
                                "cpu.metrics": {
                                    "type": "text",
                                    "value": [
                                        "percentages",
                                        "normalized_percentages"
                                    ]
                                }
                            },
                            "enabled": true
                        },
                        {
                            "compiled_stream": {
                                "period": "10s",
                                "diskio.include_devices": null,
                                "metricsets": [
                                    "diskio"
                                ]
                            },
                            "data_stream": {
                                "type": "metrics",
                                "dataset": "system.diskio"
                            },
                            "id": "system/metrics-system.diskio-e7869804-6608-452e-8f33-59b008108f57",
                            "vars": {
                                "period": {
                                    "type": "text",
                                    "value": "10s"
                                },
                                "diskio.include_devices": {
                                    "type": "text",
                                    "value": []
                                }
                            },
                            "enabled": true
                        },
                        {
                            "compiled_stream": {
                                "period": "10s",
                                "metricsets": [
                                    "filesystem"
                                ],
                                "processors": [
                                    {
                                        "drop_event.when.regexp": {
                                            "system.filesystem.mount_point": "^/(sys|cgroup|proc|dev|etc|host|lib|snap)($|/)"
                                        }
                                    }
                                ]
                            },
                            "data_stream": {
                                "type": "metrics",
                                "dataset": "system.filesystem"
                            },
                            "id": "system/metrics-system.filesystem-e7869804-6608-452e-8f33-59b008108f57",
                            "vars": {
                                "period": {
                                    "type": "text",
                                    "value": "10s"
                                },
                                "processors": {
                                    "type": "yaml",
                                    "value": "- drop_event.when.regexp:\n    system.filesystem.mount_point: ^/(sys|cgroup|proc|dev|etc|host|lib|snap)($|/)\n"
                                }
                            },
                            "enabled": true
                        },
                        {
                            "compiled_stream": {
                                "period": "1m",
                                "metricsets": [
                                    "fsstat"
                                ],
                                "processors": [
                                    {
                                        "drop_event.when.regexp": {
                                            "system.fsstat.mount_point": "^/(sys|cgroup|proc|dev|etc|host|lib|snap)($|/)"
                                        }
                                    }
                                ]
                            },
                            "data_stream": {
                                "type": "metrics",
                                "dataset": "system.fsstat"
                            },
                            "id": "system/metrics-system.fsstat-e7869804-6608-452e-8f33-59b008108f57",
                            "vars": {
                                "period": {
                                    "type": "text",
                                    "value": "1m"
                                },
                                "processors": {
                                    "type": "yaml",
                                    "value": "- drop_event.when.regexp:\n    system.fsstat.mount_point: ^/(sys|cgroup|proc|dev|etc|host|lib|snap)($|/)\n"
                                }
                            },
                            "enabled": true
                        },
                        {
                            "compiled_stream": {
                                "condition": "${host.platform} != 'windows'",
                                "period": "10s",
                                "metricsets": [
                                    "load"
                                ]
                            },
                            "data_stream": {
                                "type": "metrics",
                                "dataset": "system.load"
                            },
                            "id": "system/metrics-system.load-e7869804-6608-452e-8f33-59b008108f57",
                            "vars": {
                                "period": {
                                    "type": "text",
                                    "value": "10s"
                                }
                            },
                            "enabled": true
                        },
                        {
                            "compiled_stream": {
                                "period": "10s",
                                "metricsets": [
                                    "memory"
                                ]
                            },
                            "data_stream": {
                                "type": "metrics",
                                "dataset": "system.memory"
                            },
                            "id": "system/metrics-system.memory-e7869804-6608-452e-8f33-59b008108f57",
                            "vars": {
                                "period": {
                                    "type": "text",
                                    "value": "10s"
                                }
                            },
                            "enabled": true
                        },
                        {
                            "compiled_stream": {
                                "period": "10s",
                                "network.interfaces": null,
                                "metricsets": [
                                    "network"
                                ]
                            },
                            "data_stream": {
                                "type": "metrics",
                                "dataset": "system.network"
                            },
                            "id": "system/metrics-system.network-e7869804-6608-452e-8f33-59b008108f57",
                            "vars": {
                                "period": {
                                    "type": "text",
                                    "value": "10s"
                                },
                                "network.interfaces": {
                                    "type": "text",
                                    "value": []
                                }
                            },
                            "enabled": true
                        },
                        {
                            "compiled_stream": {
                                "process.include_top_n.by_memory": 5,
                                "period": "10s",
                                "processes": [
                                    ".*"
                                ],
                                "process.include_top_n.by_cpu": 5,
                                "process.cgroups.enabled": false,
                                "process.cmdline.cache.enabled": true,
                                "metricsets": [
                                    "process"
                                ],
                                "process.include_cpu_ticks": false
                            },
                            "data_stream": {
                                "type": "metrics",
                                "dataset": "system.process"
                            },
                            "id": "system/metrics-system.process-e7869804-6608-452e-8f33-59b008108f57",
                            "vars": {
                                "process.include_top_n.by_memory": {
                                    "type": "integer",
                                    "value": 5
                                },
                                "period": {
                                    "type": "text",
                                    "value": "10s"
                                },
                                "processes": {
                                    "type": "text",
                                    "value": [
                                        ".*"
                                    ]
                                },
                                "process.include_top_n.by_cpu": {
                                    "type": "integer",
                                    "value": 5
                                },
                                "process.cgroups.enabled": {
                                    "type": "bool",
                                    "value": false
                                },
                                "process.cmdline.cache.enabled": {
                                    "type": "bool",
                                    "value": true
                                },
                                "process.env.whitelist": {
                                    "type": "text",
                                    "value": []
                                },
                                "process.include_cpu_ticks": {
                                    "type": "bool",
                                    "value": false
                                }
                            },
                            "enabled": true
                        },
                        {
                            "compiled_stream": {
                                "period": "10s",
                                "metricsets": [
                                    "process_summary"
                                ]
                            },
                            "data_stream": {
                                "type": "metrics",
                                "dataset": "system.process.summary"
                            },
                            "id": "system/metrics-system.process.summary-e7869804-6608-452e-8f33-59b008108f57",
                            "vars": {
                                "period": {
                                    "type": "text",
                                    "value": "10s"
                                }
                            },
                            "enabled": true
                        },
                        {
                            "compiled_stream": {
                                "period": "10s",
                                "metricsets": [
                                    "socket_summary"
                                ]
                            },
                            "data_stream": {
                                "type": "metrics",
                                "dataset": "system.socket_summary"
                            },
                            "id": "system/metrics-system.socket_summary-e7869804-6608-452e-8f33-59b008108f57",
                            "vars": {
                                "period": {
                                    "type": "text",
                                    "value": "10s"
                                }
                            },
                            "enabled": true
                        },
                        {
                            "compiled_stream": {
                                "period": "10s",
                                "metricsets": [
                                    "uptime"
                                ]
                            },
                            "data_stream": {
                                "type": "metrics",
                                "dataset": "system.uptime"
                            },
                            "id": "system/metrics-system.uptime-e7869804-6608-452e-8f33-59b008108f57",
                            "vars": {
                                "period": {
                                    "type": "text",
                                    "value": "10s"
                                }
                            },
                            "enabled": true
                        }
                    ],
                    "vars": {
                        "system.hostfs": {
                            "type": "text",
                            "value": ""
                        }
                    },
                    "type": "system/metrics",
                    "enabled": true
                }
            ],
            "revision": 2,
            "created_at": "2022-08-03T08:15:00.223Z",
            "created_by": "admin",
            "updated_at": "2022-08-03T08:15:27.739Z",
            "updated_by": "admin"
        },
...
```

**Comment by mdelapenya** [comment]
@jlind23 I've detected that the system integration scenario fails if/when the list of policies obtained from `/api/fleet//package_policies` does not contain the policy that was created for the current scenario. This policy is created sending a POST request to `/api/fleet/agent_policies` with this JSON body payload:

```golang
	reqBody := `{
		"description": "Test policy ` + policyUUID + `",
		"namespace": "default",
		"monitoring_enabled": ["logs", "metrics"],
		"name": "test-policy-` + policyUUID + `"
	}`
```

And it could be the case that a scenario failed in a build for a given OS and a value (core, cpu, diskio, fsstat...), and the same scenario passed for the same OS but another value (using same code), but the very same scenario passed in a different OS.

File: e2e/_suites/fleet/system_integration.go [hunk 1]
```diff
@@ -15,25 +15,13 @@ import (
...
-	"github.com/elastic/e2e-testing/internal/common"
-	"github.com/elastic/e2e-testing/internal/deploy"
 	"github.com/elastic/e2e-testing/internal/kibana"
 	"github.com/elastic/e2e-testing/internal/utils"
 	"github.com/google/uuid"
 	"github.com/pkg/errors"
 	log "github.com/sirupsen/logrus"
 )
 
-func (fts *FleetTestSuite) getAgentOSData() (string, error) {
```
Comment (mdelapenya) [inline e2e/_suites/fleet/system_integration.go:27]: Remove unused method

**Comment by jlind23** [comment]
@mdelapenya trying to catch up here. You mean that the policy created for this given scenario is not available/present when requesting the policies list from the API right? Would that mean that as you said the tests ran too fast and fleet was not able to retrieve the policy in time?

**Comment by mdelapenya** [comment]
> @mdelapenya trying to catch up here. You mean that the policy created for this given scenario is not available/present when requesting the policies list from the API right? Would that mean that as you said the tests ran too fast and fleet was not able to retrieve the policy in time?

Yes, we are seeing that behavior when the system integration scenario fails. In this PR we are polling [this diagram](https://github.com/elastic/e2e-testing/pull/2903#issuecomment-1202867885) for a period of time: 

1. list the policies
  1.1. found the policy by ID
    1.1.1. retrieve the policy from the API
  1.2 policy not found by ID --> repeat until maxTimeOut

With this PR we are adding logs so that, when a failure happens, we have the list of the existing policies. We thought that adding the poll would help, but I think it's not a matter of waiting for the policy to be in the list, so not really sure about what's going on for the policy to not be present in the list

**Review by jlind23 — APPROVED** [review]

**Comment by jlind23** [comment]
@mdelapenya lets try adding the logs first in order to have more informations when a failure happens.

**Comment by mdelapenya** [comment]
I'll be running this PR few times to get more information. Thanks @jlind23 for your help!

**Review by cachedout — APPROVED** [review]

**Review by pazone — APPROVED** [review]

**Pull request merged**